rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+---
+openapi: 3.0.3
+info:
+  title: Apache Iceberg REST Catalog API
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: 1.0.0
+  description:
+    Defines the specification for the first version of the REST Catalog API. Implementations should support both Iceberg table specs v1 and v2, with priority given to v2.
+servers:
+  - url: http://127.0.0.1:1080
+    description: URL Used for Mock-Server Unit Tests
+# All routes are currently configured using an Authorization header.
+security:
+  - BearerAuth: []
+paths:
+  /v1/config:
+    get:
+      tags:
+        - Configuration API
+      summary: List all catalog configuration settings
+      operationId: getConfig
+      description:
+        All REST catalogs will be initialized by calling this route. This route
+        will return at least the minimum necessary metadata to initialize the
+        catalog. Optionally, it can also include server-side specific overrides.
```

Comment:
This states that the settings are overrides. But what about defaults? Should we have a way to specify whether a setting _must_ be used or is just a default?

Reply:
That's a good question.

I think that users should be able to (and need to be able to) specify configuration values on their end. The server should _at least_ be able to specify defaults that the user can possibly override.

I'd like for it to be able to specify values that must be used (e.g. overriding any user supplied setting), but that might not be needed just yet.

If we did want to allow that, maybe the server could specify configs as a JSON blob, and specify if the server value must be used? (Note that I grabbed this config key randomly, so the value layout is more important).

Using a JSON object for the configuration allows us to slip in more parameters that way overtime.
```
{ 
   "config1": { .... },
   "spark.sql.iceberg.vectorization.enabled": { "default": true, "allow_overrides": true } },
 }
 ```